Flaky-test: ManagedLedgerBkTest.asyncMarkDeleteAndClose

[example failure](https://github.com/apache/pulsar/runs/7449515309?check_suite_focus=true)

```
Error:  Tests run: 25, Failures: 12, Errors: 0, Skipped: 13, Time elapsed: 7.366 s <<< FAILURE! - in org.apache.bookkeeper.mledger.impl.ManagedLedgerBkTest
  Error:  asyncMarkDeleteAndClose(org.apache.bookkeeper.mledger.impl.ManagedLedgerBkTest)  Time elapsed: 0.129 s  <<< FAILURE!
  java.lang.AssertionError: expected [false] but found [true]
  	at org.testng.Assert.fail(Assert.java:99)
  	at org.testng.Assert.failNotEquals(Assert.java:1037)
  	at org.testng.Assert.assertFalse(Assert.java:67)
  	at org.testng.Assert.assertFalse(Assert.java:77)
  	at org.apache.bookkeeper.mledger.impl.ManagedLedgerBkTest.asyncMarkDeleteAndClose(ManagedLedgerBkTest.java:338)
  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
  	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:132)
  	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:45)
  	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:73)
  	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:11)
  	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
  	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
  	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
  	at java.base/java.lang.Thread.run(Thread.java:833)
```

<!-- optionally provide the full stacktrace ->
<details>
<summary>Full exception stacktrace</summary>
<code><pre>
full exception stacktrace here
</pre></code>
</details>

--- a/managed-ledger/src/test/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerBkTest.java
+++ b/managed-ledger/src/test/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerBkTest.java
@@ -330,10 +330,10 @@ public class ManagedLedgerBkTest extends BookKeeperClusterTestCase {
             }, null);
         }
 
+        counter.await();
+
         cursor.close();
         ledger.close();
-
-        counter.await();
 
         assertFalse(gotException.get());
     }

